compiler/rustc_infer/src/infer/outlives/obligations.rs

```diff
@@ -392,13 +392,16 @@ where
         // the problem is to add `T: 'r`, which isn't true. So, if there are no
         // inference variables, we use a verify constraint instead of adding
         // edges, which winds up enforcing the same condition.
-        let is_opaque = alias_ty.kind(self.tcx) == ty::Opaque;
+        //
+        // Note that this is not here for correctness; it solely affects
```

Comment:
This is not fucking true lmao

Reply:
I'll fix the comment anyways

Reply:
please do :blush: 

my understanding is that this weakens region inference for the sake of diagnostics, but does not do so for opaques because there the diagnostics benefits aren't worth the loss of expressiveness? :thinking:

Like, the "proper" handling would be to always use `args_must_outlive` if there are no declared bounds, but track the origin of the arg outlives to make sure we still talk about `T::Item`?

Reply:
> my understanding is that this weakens region inference for the sake of diagnostics

This block strengthens region inference when the type is opaque or if it has region variables. For opaques, it's to avoid bad diagnostics like `where {opaque}: 'a`, and when the type has region variables we need to add constraints to the region graph to ensure that a type test doesn't fail.

Reply:
I have no idea what you are arguing here. You are simply stating exactly what exists + I wrote in the comment. Please re-read the code 🤔

Reply:
The only thing that I can gather from the comment is that the last comment is nitpicking "correcntess" vs "expressiveness". Is that correct?

Reply:
Actually, I'm gonna go further and just remove my addition to the comment here. The existing comment explained the situation enough, and I don't really wanna argue any of this further.